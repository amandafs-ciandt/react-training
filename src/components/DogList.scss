.dog-list {
  padding: 1rem;

  &__content {
    height: 13rem;
    overflow: auto;
  }

  &__item {
    cursor: pointer;
    border-bottom: 1px solid #f0f0f0;
  }
}
