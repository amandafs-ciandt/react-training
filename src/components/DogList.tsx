import React from 'react';
import { useList } from 'effector-react';

import { filtered } from '../effector/store';
import { selectDog } from '../effector/event';

import { List } from 'antd';
import './DogList.scss';

import DogItem from './DogItem';

const DogList = () => {
  const dogsList = useList(filtered, (dog, itemIndex) => {
    return (
      <div className='dog-list__item' onClick={() => selectDog(dog)}>
        <DogItem key={itemIndex} dog={dog} />
      </div>
    );
  });

  return (
    <div className='dog-list'>
      <List className='dog-list__content' itemLayout='horizontal'>
        {dogsList}
      </List>
    </div>
  );
};

export default DogList;
