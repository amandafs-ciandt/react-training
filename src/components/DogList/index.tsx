import React from "react";
import { useStore } from "effector-react";

import { filtered } from "../../effector/store";
import { selectDog } from "../../effector/event";

import { List } from "antd";
import "./DogList.scss";

import DogItem from "../DogItem";

const DogList = () => {
  const dogsList = useStore(filtered);

  return (
    <div className="dog-list">
      <List className="dog-list__content" itemLayout="horizontal">
        {dogsList.map((dog, itemIndex) => (
          <div
            key={itemIndex}
            className="dog-list__item"
            onClick={() => selectDog(dog)}
          >
            <DogItem dog={dog} />
          </div>
        ))}
      </List>
    </div>
  );
};

export default DogList;
