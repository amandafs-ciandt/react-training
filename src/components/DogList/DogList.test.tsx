import React from "react";
import * as effector from "effector-react";
import { shallow, ShallowWrapper, ReactWrapper, mount } from "enzyme";

import DogList from ".";

const setUp = () => {
  return shallow(<DogList />);
};

const dogs = [
  {
    breed: "bulldog",
    image: "https://images.dog.ceo/breeds/bulldog-french/n02108915_4474.jpg",
    scolded: 3,
  },
  {
    breed: "collie",
    image: "https://images.dog.ceo/breeds/collie-border/n02106166_1460.jpg",
    scolded: 0,
  },
  {
    breed: "spaniel",
    image: "https://images.dog.ceo/breeds/spaniel-blenheim/n02086646_2815.jpg",
    scolded: 1,
  },
];

const spy = jest.spyOn(effector, "useStore");
spy.mockReturnValue(dogs);

fdescribe("DogList", () => {
  let dogListComponent: ShallowWrapper | ReactWrapper;

  beforeEach(() => {
    dogListComponent = setUp();
  });

  it("should work", () => {
    expect(dogListComponent).toMatchSnapshot();
  });

  it("should render the entire list", () => {
    const listItems = dogListComponent.find("DogItem");

    expect(listItems.length).toEqual(dogs.length);
  });

  it("should call selectDog when clicking on an item of the list", () => {
    dogListComponent = mount(<DogList />);

    const firstListItem = dogListComponent.find(".dog-list__item").first();
    firstListItem.simulate("click");

    expect(true).toBeTruthy();
  });
});
