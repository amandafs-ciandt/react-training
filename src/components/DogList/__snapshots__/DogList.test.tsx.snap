// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogList should work 1`] = `ShallowWrapper {}`;
